Objects/typeobject.c

```diff
@@ -6036,15 +6036,17 @@ object_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
             return NULL;
         }
         comma_w_quotes_sep = PyUnicode_FromString("', '");
-        joined = PyUnicode_Join(comma_w_quotes_sep, sorted_methods);
-        method_count = PyObject_Length(sorted_methods);
-        Py_DECREF(sorted_methods);
-        if (joined == NULL)  {
+        if (comma_w_quotes_sep) {
```

Comment:
This is a bit confusing to read because it differs from the common pattern of handling the error case in an if-statement with an early return.

GitHub is also showing warnings because `joined` is not defined when `comma_w_quotes_sep` is `NULL`.

Reply:
I thought `joined` was initialised, but I guess it's obviously not. And yeah, after I shuffled things one more time the positive check is now more out of place (but the indentation is flatter), so flipping it and repeating decrefs is probably better.